pkg/kubectl/cmd/cmd.go

```diff
@@ -331,7 +332,16 @@ func DefaultClientConfig(flags *pflag.FlagSet) clientcmd.ClientConfig {
 
 func checkErr(err error) {
```

Comment:
Example:

```
func (f *Factory) NewCmdGet(out io.Writer) *cobra.Command {
	cmd := &cobra.Command{
		Use:     "get [(-o|--output=)json|yaml|template|...] <resource> [<id>]",
		Short:   "Display one or many resources",
		Long:    get_long,
		Example: get_example,
		Run: func(cmd *cobra.Command, args []string) {
			RunGet(f, out, cmd, args)
		},
	}
        // ...
	return cmd
}
```

Should probably be

```
func (f *Factory) NewCmdGet(out io.Writer) *cobra.Command {
	cmd := &cobra.Command{
		Use:     "get [(-o|--output=)json|yaml|template|...] <resource> [<id>]",
		Short:   "Display one or many resources",
		Long:    get_long,
		Example: get_example,
		Run: func(cmd *cobra.Command, args []string) {
			err := RunGet(f, out, cmd, args)
			checkErr(err)
		},
	}
        // ...
	return cmd
}
```

There's no reason our code has to run inside of the Run() method.  That would then allow us to a) unit test RunGet (which we need to be able to do) and b) centralize all exit handling.

----- Original Message -----
> > @@ -331,7 +332,16 @@ func DefaultClientConfig(flags *pflag.FlagSet)
> > clientcmd.ClientConfig {
> >  
> >  func checkErr(err error) {
> 
> @smarterclayton, apologies if I'm being dense, but what is the pattern you
> are suggesting we use here? Log error and then return? Or are you suggesting
> that we fork/upstream commander's Run function to return an error and push
> error handling into the command framework?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/5016/files#r25807240